client/common/util.js

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017 The Absolute Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * common util class
+ */
+export default class Util {
+  /**
+   * get server URL
+   */
+  static getServerURL() {
+    const HOST = self.location.hostname;
+    const PORT = self.location.port;
+    const SERVER_URL = `//${HOST}:${PORT}`;
+    return SERVER_URL;
+  }
+
+  /**
+   * fetch request operation
+   */
+  static fetchRequest(request) {
+    return new Promise(function(resolve, reject) {
+      fetch(request).then(function(response) {
```

Comment:
The fetch() API is already promise-based API[1]. Why do we need this wrapper?

[1] https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Basic_concepts